board.h

```diff
@@ -0,0 +1,83 @@
+#pragma once
+#include <string>
+#include <vector>
+#include <fstream>
+
+// abstract parent class for both snakes and ladders
+// they just modify the positions differently, that's why I have used virtual function for it
+class Snake_Ladder
+{
+protected:
+    int low, high;
+
+public:
+    Snake_Ladder(int l, int h) : low(l), high(h) {}
```

Comment:
avoid l, h. use proper variable names

Reply:
I thought it's okay for them to be used in such places, where it doesn't cause problem to the readability or understandability of the code. But will work on it, thanks!